### ISSUE TITLE
binary-compat: `opendir()` fails under binary compatibility mode

### ISSUE THREAD

**Issue by razvand** [unikraft/unikraft#497]
### Describe the bug

When running a static PIE executable under binary compatibility mode, using [`run-app-elfloader`](https://github.com/unikraft/run-app-elfloader), it errs when using `opendir()`:

```
opendir: Not a directory
```

### Steps to reproduce

Use [`run-app-elfloader`](https://github.com/unikraft/run-app-elfloader) to run a simple static PIE binary that uses `opendir()` such as [this one](https://github.com/unikraft/static-pie-apps/pull/11). Use the command below, assuming the [`run-app-elfloader` repository](https://github.com/unikraft/run-app-elfloader) and the [`static-pie-apps` repository](https://github.com/unikraft/static-pie-apps/) are cloned in the same directory:

```
$ ./run_elfloader -r ../static-pie-apps/small/dir/rootfs/  ../static-pie-apps/small/dir/dir
```

The output shows the Unikraft boot message together with the error message on `opendir()`:

```
[...]
Powered by
o.   .o       _ _               __ _
Oo   Oo  ___ (_) | __ __  __ _ ' _) :_
oO   oO ' _ `| | |/ /  _)' _` | |_|  _)
oOo oOO| | | | |   (| | | (_) |  _) :_
 OoOoO ._, ._:_:_,\_._,  .__,_:_, \___)
            Mimas 0.7.0~1e17a9c8-custom
opendir: Not a directory
```

To build your own setup, including Unikraft and aim to fix this bug, see [these instructions](https://unikraft.org/community/hackathons/2022-05-lyon/bincompat/#00-setup).

### Expected behavior

The simple program should start under binary compatibility mode and list folder contents in the root directory of the filesystem (passed using the `-r ...` option to [`run-app-elfloader`](https://github.com/unikraft/run-app-elfloader).

### Which architectures were you using or does this bug affect?

x86_64

### Which platforms were you using or does this bug affect?

kvm

**Comment by razvand** [comment]
I found the root-case of the issue. I will submit a PR.

**Comment by razvand** [comment]
Fixed by #670
